jung-visualization/src/main/java/edu/uci/ics/jung/visualization/BaseVisualizationModel.java

```diff
@@ -107,19 +107,21 @@ public BaseVisualizationModel(
     }
   }
 
-  private LayoutModel<N, Point2D> createLayoutModel(Network<N, E> network, Dimension layoutSize) {
-    // the default
-    SpatialSupport spatialSupport = SpatialSupport.QUAD_TREE;
-    String spatialSupportProperty = System.getProperty("layout.model");
+  private SpatialSupport getSpatialSupportPreference() {
+    String spatialSupportProperty = System.getProperty("spatial.support");
```

Comment:
I'm OK with this being a system property while we're developing, if that makes things more convenient for you, but I don't want to release JUNG that way.

If you want to keep it as a property, please file an issue to refactor the code to get rid of this property (either by removing the option, or by allowing users to specify it programmatically), and add a comment here:

// TODO(#issue): remove this property

so that we know where we need to start.

Reply:
You should give that some careful thought.
This is a visualization module based on swing which is highly customizable using properties. 
https://docs.oracle.com/javase/8/docs/api/javax/swing/plaf/synth/doc-files/componentProperties.html
There is so much in our visualization code that users could put into properties files and give themselves a great amount of flexibility to customize code *without recompiling*. 
Javafx will probably be the same, with properties in style sheets.

Reply:
I'm fine with having that discussion, but we (JUNG) have not done that with literally any other setting that I'm aware of, so I think that if you want to introduce that practice that we need to in fact have that discussion (and we shouldn't try to have it in the middle of a code review which I'm trying not to block ;) ).

Reply:
...then don't put ultimatums in your code review. :)
I know you don't really mean it that way. We have many years of discussing things like 'what does implements DirectedGraph really mean?'. It's always been productive.

In fact, when a user makes a jung visualization, they can customize *everything* about it *except* the jung part with properties. We've always made users programmatically climb down the path to some function in some renderer in some render context instead of just letting them do this:

jung.defaultEdgeShape=QuadCurve
jung.defaultNodeColor=red
jung.defaultPickedNodeColor=yellow
jung.defaultNodeLabelPosition=CNTR

...and so on
in a file called jung.properties

This was a historically huge oversight on my part.

Reply:
As I said, I'm happy to have that discussion, I just don't think that a code review is the place to do it.  It makes it harder for other people to participate, and it makes it harder for us to find it later (especially because of the really obnoxious way that GitHub's review comments work).  I think that I see things somewhat differently, but I'm certainly willing to be convinced.

And whatever decision we make about this sort of configuration, I agree that we should try to be reasonably uniform about it.

So please do file that issue so that we can work this out.  Thanks.